core/frontend/src/main/scala/io/udash/properties/single/Property.scala

```diff
@@ -69,7 +69,19 @@ trait ReadableProperty[A] {
   /** Registers listener which will be called on value change. */
   def listen(l: A => Any): Registration = {
     listeners += l
-    new PropertyRegistration(listeners, l)
+    new SetRegistration(listeners, l)
+  }
+
+  /** Registers listener which will be called on the next value change. This listener will be fired only once. */
+  def listenOnce(l: A => Any): Registration = {
+    val wrapper: A => Any = new Function1[A, Any] {
```

Comment:
Does this require the full anonymous class syntax?

Reply:
I need `this` reference here